src/move_embargo_args.py

```diff
@@ -49,35 +48,32 @@ def parse_args():
     collections = namespace.collections
 
     # Dataset to move
-    # dataID must include
-    if not [x for x in (namespace.instrument, namespace.detector, namespace.band) if x is None]:
-        dataId = {'instrument': namespace.instrument, 'detector': namespace.detector,
-                  'band': namespace.band}
-    elif not [x for x in (namespace.instrument, namespace.detector, namespace.band, namespace.exposure) if x is None]:
-        dataId = {'instrument': namespace.instrument, 'detector': namespace.detector,
-                  'band': namespace.band, 'exposure': namespace.exposure}
-    else:
-        dataId = {'instrument': namespace.instrument}
+    dataId = {'instrument': namespace.instrument}
 
     # Define embargo period
-    embargo_period = astropy.time.TimeDelta(namespace.embargodays, format='jd')
-    now = astropy.time.Time.now()
-    int_now = int(now.datetime.strftime("%Y%m%d"))
+    embargo_period = astropy.time.TimeDelta(namespace.embargohours*3600., format='jd')
+    if namespace.nowtime != 'now':
+        now = astropy.time.Time(namespace.nowtime, scale='tai', format='iso')
+    else:
+        now = astropy.time.Time.now()
+    timespan_embargo = Timespan(now - embargo_period, now)
 
     # The Dimensions query
-    # If now - observation_end_time_in_embargo > embargo period : move
+    # If (now - embargo period, now) does not overlap
+    # with observation time interval: move
     # Else: don't move
     # Save data Ids of these observations into a list
     after_embargo = []
 
     for i, dt in enumerate(registry.queryDimensionRecords('exposure', dataId=dataId, datasets=datasetType,
```

Comment:
If you're not using `i`, you can drop the `enumerate()`.  And if you're just appending these to a list, it's even simpler to say `after_embargo = list(registry.queryDimensionRecords(...))` (or if you really want to extract the ids, `after_embargo = [dt.id for dt in registry.queryDimensionRecords(...)]`)

Reply:
I like this format and included this in the code.
We also want to test if we need the IF condition within the query. So, I am commenting it out for now.